wayat/frontend/lib/app_state/notification_state/notification_state.dart

```diff
@@ -0,0 +1,263 @@
+  void setToken(String token) {
+    NotificationServiceImpl().sendCurrentUserToken(token);
+    print(
+        '-----------------------------------------Send refreshed token to backend: $token');
+  }
+
+  /// Get FCM token
+  void getToken() {
+    messagingInstance.getToken().then((token) {
+      if (token != null) setToken(token);
+      print(
+          '-----------------------------------------Send token to backend: $token');
+    });
+    _tokenStream = messagingInstance.onTokenRefresh;
+    _tokenStream.listen(setToken);
+  }
+
+  /// Return notification with [newMessage]
+  PushNotification pushNotification(RemoteMessage newMessage) {
+    if (platformService.isWeb) {
+      // TODO: implement web notification
+    }
+
+    if (platformService.targetPlatform == TargetPlatform.android) {
+      return PushNotification(
+        title: newMessage.notification!.title ?? "",
+        body: newMessage.notification!.body ?? ""
+      );
+    }
+
+    if (platformService.targetPlatform == TargetPlatform.iOS) {
+      return PushNotification(
+        title: newMessage.data['aps']['alert']['title'],
+        body: newMessage.data['aps']['alert']['body'],
+      );
+    }
+    return PushNotification(
+      title: 'No title',
+      body: 'No body'
+    );
+  }
+
+  /// For handling notification when the app is open
+  messagingAppListener() {
+    FirebaseMessaging.onMessage.listen((RemoteMessage message) {
+      notificationInfo = pushNotification(message);
+      totalNotifications++;
+      if (notificationInfo != null) showNotification();
+    });
+  }
+
+  // For handling notification when the app is in terminated state
+  @action
+  messagingTerminatedAppListener() async {
+    RemoteMessage? initialMessage = await messagingInstance.getInitialMessage();
+    if (initialMessage != null) {
+      notificationInfo = pushNotification(initialMessage);
+      totalNotifications++;
+      notificationDetails();
+    }
+  }
+
+  /// For handling notification when the app is in background
+  /// but not terminated
+  @action
+  messagingBackgroundAppListener() async {
+    FirebaseMessaging.onMessageOpenedApp.listen((RemoteMessage message) {
+      notificationInfo = pushNotification(message);
+      totalNotifications++;
+      notificationDetails();
+    });
+  }
+
+  showNotification() {
+    showSimpleNotification(
+      Text(notificationInfo?.title ?? 'No notification title'),
+      subtitle: Text(notificationInfo?.body ?? 'No notification body'),
+      background: Colors.white,
+      foreground: Colors.black,
+      leading: NotificationBadge(),
+      trailing: OutlinedButton(
+        style: OutlinedButton.styleFrom(
+          backgroundColor: Colors.white,
+          foregroundColor: Colors.black,
+          shape: RoundedRectangleBorder(
+            borderRadius: BorderRadius.circular(1000.0),
+            side: const BorderSide(color: Colors.black, width: 1),
+          ),
+        ),
+        onPressed: (() {
```

Comment:
If this is the dismiss button please remove it, if not remove the TODO comment

Reply:
Done